custom/ewsghana/reports/stock_levels_report.py

```diff
@@ -288,7 +287,8 @@ def calculate_weeks_remaining(stock_state, date):
 
         rows = {}
         for state in stock_states:
-            product_name = Product.get(state.product_id).name
+            product = Product.get(state.product_id)
+            product_name = product.name + ' (%s)' % product.code
```

Comment:
if you're just getting the names and codes then `SQLProduct` will likely be faster. it's also generally better to load them from the DB once as opposed to one at a time.